src/public_witness.rs

```diff
@@ -0,0 +1,64 @@
+use acvm::acir::{
+    circuit::Circuit,
+    native_types::{Witness, WitnessMap},
+};
+use wasm_bindgen::{prelude::wasm_bindgen, JsValue};
+
+use crate::JsWitnessMap;
+
+fn extract_indices(witness_map: &WitnessMap, indices: Vec<Witness>) -> Result<WitnessMap, String> {
+    let mut extracted_witness_map = WitnessMap::new();
+    for witness in indices {
+        let witness_value = witness_map.get(&witness).ok_or(format!(
+            "Failed to extract witness {} from witness map. Witness not found.",
+            witness.0
+        ))?;
+        extracted_witness_map.insert(witness, *witness_value);
+    }
+    Ok(extracted_witness_map)
+}
+
+#[wasm_bindgen(js_name = getReturnWitness)]
```

Comment:
I'm not a huge fan of these names but we can bikeshed this closer to the first stable release.